src/controllers/favorite-controller.js

```diff
@@ -3,7 +3,7 @@ const NotfoundError = require('../errors/not-found-error');
 
 async function deleteFavorite(req, res, next){
   const { productId }  = req.params;
-  const userId = res.userData._id;
+  const userId = req.userData._id;
```

Comment:
for example here, you changed the payload of token, there won't be req.userData._id after your changes